### ISSUE TITLE
October Iteration Plan

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#13342]
This plan captures what we work on during October. We will ship **early November**.

## Endgame Schedule
- _October, 24th_ Code freeze for the endgame
- _October, 30th_ Endgame done

##### Verification
- [x] Verification ([wiki](https://github.com/Microsoft/vscode/wiki/Issue-Tracking#verification)): since there are quite some old bugs to verify we verify by age, youngest first. The queries are:
  - [x] [Verification needed](https://github.com/Microsoft/vscode/issues?q=is%3Aissue+-label%3Averified+is%3Aclosed+label%3Averification-needed)
  - [x] [October Bugs](https://github.com/Microsoft/vscode/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Abug%20-label%3Averified%20is%3Aclosed%20-label%3Aduplicate%20-label%3Ainvalid%20closed%3A%3E%3D2016-10-01%20-label%3Aas-designed)

##### Monday
- [x] Code freeze at 5pm PT
- [x] Ensure we have a green build on all platforms
- [x] All test items contain sufficiently comprehensive test descriptions by 6pm PT

##### Tuesday
- [x] Test build starts at 7am CET / 10pm PT on Monday
- [x] Test plan ready by 8am CET / 11pm PT on Monday
- [x] Testing

##### Wednesday
- [x] Testing
- [x] Remind team members to assign issues that they intend to fix to the October milestone
- [x] Fixing (self-assigned, milestone assigned)
- [x] Verification (queries above)

##### Thursday
- [x] Fixing (scrutiny sets in - major bugs only - to be discussed in stand-up meeting, labeled as `candidate`)
- [x] Verification (queries above)
- [x] Add/update shrink-wrap files for built-in extensions if needed (see [instructions](https://github.com/Microsoft/vscode/issues/8570#issuecomment-229669456))
- [x] Update `OSSREADME.json` for built-in extensions based on differences to generated `npm-shrinkwrap.json` files if **@basero**
- [x] Run [OSS tool](https://github.com/Microsoft/vscode-distro/blob/master/distro-tools/README.md) after merging shrink-wrap findings **@bpasero**
  - _The LCA review of the ThirdPartyNotices.txt files is not needed anymore_
- [x] Check new OSS usage is entered into the [OSS registry](https://ossmsft.visualstudio.com/_apps/hub/ms.vss-oss-web.hub-oss) **@bpasero**

##### Friday
- Satellite modules/npm packages ready, version updated, smoke tested
  - [x] vscode **@jrieken**
  - [x] yo generator **@aeschli**
  - [x] vsce **@joaomoreno**
  - [x] node debug **@weinand**
  - [x] node debug2 **@roblourens**
- [x] Translation input - **@dbaeumer**
- [x] Verification (queries above)
- [x] Fixing (only critical bugs - no string changes)
- [x] All release notes updated. Release notes are collected in a file named _`1_7.md`_ in this [repo directory](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/)
  - [x] @alexandrudima
  - [x] @weinand
  - [x] @bpasero 
  - [x] @chrmarti
  - [x] @tyriar
  - [x] @dbaeumer
  - [x] @egamma
  - [x] @isidorn 
  - [x] @joaomoreno
  - [x] @jrieken
  - [x] @kieferrm 
  - [x] @aeschli
  - [x] @rebornix
  - [x] @octref
  - [x] @ramya-rao-a
  - [x] @roblourens
  - [x] @sandy081
- [x] Acknowledge pull requests in release notes. We acknowledge PRs from outside the team. 
  Use the [thankyou](https://github.com/Microsoft/vscode-tools/tree/master/thankyou) utility to create the initial version @bpasero
  - [x] @alexandrudima
  - [x] @weinand
  - [x] @bpasero 
  - [x] @chrmarti
  - [x] @tyriar
  - [x] @dbaeumer
  - [x] @egamma
  - [x] @isidorn 
  - [x] @joaomoreno
  - [x] @jrieken
  - [x] @kieferrm 
  - [x] @aeschli
  - [x] @rebornix
  - [x] @octref
  - [x] @ramya-rao-a
  - [x] @roblourens
  - [x] @sandy081
- [x] Mention notable fixes in the release notes:
  - [x] @alexandrudima
  - [x] @weinand
  - [x] @bpasero 
  - [x] @chrmarti
  - [x] @tyriar
  - [x] @dbaeumer
  - [x] @egamma
  - [x] @isidorn 
  - [x] @joaomoreno
  - [x] @jrieken
  - [x] @kieferrm 
  - [x] @aeschli
  - [x] @rebornix
  - [x] @octref
  - [x] @ramya-rao-a
  - [x] @roblourens
  - [x] @sandy081
- [Smoketest](https://github.com/Microsoft/vscode/wiki/Smoke-Test)
  - [x] Windows - **@bpasero**
  - [x] OS X - **@isidorn**
  - [x] Linux - **@joaomoreno**

##### Friday/Monday
- When done fixing/verifying and there are changes since last build at the end of day PT
  - [x] Trigger new insider build and publish it manually **@bpasero**
- [x] Pause scheduled `insider` builds **@bpasero**
- [x] Branch code to `release/<x.y> **@bpasero**
- [x] Announce master is open for business **@bpasero**
- [x] Polish release notes **@gregvanl**

##### Wednesday
- [x] Publish `Insider` with hand-picked and reviewed candidate fixes **@bpasero**
- [Smoketest](https://github.com/Microsoft/vscode/wiki/Smoke-Test)
  - [x] Windows - **@chmarti**
  - [x] OS X - **@roblourens**
  - [x] Linux - **@tyriar**

##### Thursday/Monday
- [x] Build stable for all platforms **@bpasero**
- [x] Sanity check of installable bits
  - [x] Windows
    - [x] signed installer **@bpasero**
    - [x] zip **@bpasero**
  - [x] OS X - **@bpasero**
  - [x] Linux
    - [x] deb package 32-bit **@tyriar**
    - [x] deb package 64-bit **@bpasero**
    - [x] rpm package 64-bit **@tyriar**
    - [x] rpm package 32-bit **@tyriar**
    - [x] archives **@bpasero**
- [x] Publish website **@gregvanl**
- [x] Release [stable](https://vscode-builds.azurewebsites.net/stable) **@bpasero**
- [x] Add a git tag to `HEAD` of `release/1.7` in format `1.7.z` **@bpasero**
- [ ] Twitter announcement **@chrisdias**
- [x] Enable scheduled `insider` builds **@bpasero**

## Plan Items

We are now using the new [Project](https://github.com/blog/2256-a-whole-new-github-universe-announcing-new-tools-forums-and-features) support to track the work planned for [October](https://github.com/Microsoft/vscode/projects/2).

The columns in the project mean:
- `Backlog` - the backlog for the iteration
- `Planned to Start Current week` - the work planned to start in the current week of the iteration
- `In Progress` - the item is actively being worked on
- `Done` - the work item is done and there is a corresponding test plan item.

**Comment by mi-hol** [comment]
Currently the github 'projects' page shows 'no due date'. 
Community member are hence a bit in the dark and few will find this plan.
Would the dates published above not fit better on 'projects' page?

**Comment by egamma** [comment]
@mi-hol thanks for your feedback, the example link you provide below is not accessible to me.

Here is how I think about it:
- we still use milestones on issues for when we plan to deliver an item. Milestones have a due date (it was an oversight to not assign a milestone date for October).
- we use 'Projects' to track the progress on these issues we plan for October, i.e., Planned this week, in progress, done, deferred).
- we use Cards in projects for for things that are under discussion and haven't hardened yet as issues.

Bottom line if you are not interested in the progress on an Issue as an external user, then you can just looks at the milestone of the issue and you can also look at the milestones page. However, the milestone page get difficult to digest when it has over 100 items. See the [October Milestone](https://github.com/Microsoft/vscode/milestone/29) as an example.

**Comment by mi-hol** [comment]
@egamma thanks for your fast response. I've verified the supplied link and it doesn't work when clicked inside the issue for me either but copied into a browser it does. Weird that the link inside a GitHub issue doesn't work (maybe a GitHub bug).  URL: https://github.com/serverless/serverless/milestone/15
- Assigning a due date was addressing my comment (note: 'November 2016' milestone lacks it again)

**Comment by egamma** [comment]
Closing 1.7 has shipped

**Comment by mrmlnc** [comment]
Hello, @egamma,

Small clarification: VS Code team has plans to publish recovery release? Because now we have broken support `JavaScript React` (#14962) and `Less/SCSS` comments (#15013).

**Comment by egamma** [comment]
@mrmlnc thanks for pointing this out, added these issues as candidates to the October recovery build.
